tests/su3_test.cpp

```diff
@@ -191,57 +177,66 @@ int main(int argc, char **argv)
-    printfQuda("Total time for STOUT = %g secs\n", time0);
-    break;
+  // Typical values for
+  // APE: alpha=0.6
+  // Stout: rho=0.1
+  // Over Improved Stout: rho=0.08, epsilon=-0.25
+  //
+  // Typically, the user will use smearing for Q charge data only, so
+  // we hardcode to compute Q only and not the plaquette. Users may
+  // of course set these as they wish.  SU(N) projection su_project=true is recommended.
+  QudaGaugeObservableParam *obs_param = new QudaGaugeObservableParam[gauge_smear_steps / measurement_interval + 1];
+  for (int i = 0; i < gauge_smear_steps / measurement_interval + 1; i++) {
+    obs_param[i] = newQudaGaugeObservableParam();
+    obs_param[i].compute_plaquette = QUDA_BOOLEAN_FALSE;
+    obs_param[i].compute_qcharge = QUDA_BOOLEAN_TRUE;
+    obs_param[i].su_project = su_project ? QUDA_BOOLEAN_TRUE : QUDA_BOOLEAN_FALSE;
+  }
 
-    // Topological charge routines
-    //---------------------------------------------------------------------------
-  case 2:
-    // Over-Improved STOUT
-    // start the timer
-    time0 = -((double)clock());
-    performOvrImpSTOUTnStep(smear_steps, stout_smear_rho, stout_smear_epsilon, measurement_interval);
-    // stop the timer
-    time0 += clock();
-    time0 /= CLOCKS_PER_SEC;
-    printfQuda("Total time for Over Improved STOUT = %g secs\n", time0);
+  // We here set all the problem parameters for all possible smearing types.
+  QudaGaugeSmearParam smear_param = newQudaGaugeSmearParam();
+  smear_param.smear_type = gauge_smear_type;
+  smear_param.n_steps = gauge_smear_steps;
+  smear_param.meas_interval = measurement_interval;
+  smear_param.alpha = gauge_smear_alpha;
+  smear_param.rho = gauge_smear_rho;
+  smear_param.epsilon = gauge_smear_epsilon;
+
+  time0 = -((double)clock()); // start the timer
+  switch (smear_param.smear_type) {
+  case QUDA_GAUGE_SMEAR_APE:
+  case QUDA_GAUGE_SMEAR_STOUT:
+  case QUDA_GAUGE_SMEAR_OVRIMP_STOUT: {
+    performGaugeSmearQuda(&smear_param, obs_param);
     break;
-  case 3:
-    // Wilson Flow
-    // Start the timer
-    time0 = -((double)clock());
-    performWFlownStep(wflow_steps, wflow_epsilon, measurement_interval, wflow_type);
-    // stop the timer
-    time0 += clock();
-    time0 /= CLOCKS_PER_SEC;
-    printfQuda("Total time for Wilson Flow = %g secs\n", time0);
+  }
+
+    // Here we use a typical use case which is different from simple smearing in that
+    // the user will want to compute the plaquette values to compute the gauge energy.
+  case QUDA_GAUGE_SMEAR_WILSON_FLOW:
+  case QUDA_GAUGE_SMEAR_SYMANZIK_FLOW: {
+    for (int i = 0; i < gauge_smear_steps / measurement_interval + 1; i++) {
+      obs_param[i].compute_plaquette = QUDA_BOOLEAN_TRUE;
+    }
+    performWFlowQuda(&smear_param, obs_param);
     break;
-  default: errorQuda("Undefined test type %d given", test_type);
   }
+  default: errorQuda("Undefined gauge smear type %d given", smear_param.smear_type);
+  }
+
+  time0 += clock(); // stop the timer
```

Comment:
Use the `host_timer_t` class for doing this kind of timing.   Look in dslash_test_util.h for an example.

Reply:
8c6db1f5c705446dfb5fed55428c694edc41e73e